### PR TITLE
[spaceship] add methods to rename and disable/enable devices

### PR DESCRIPTION
**Pull request by nekrich**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I see several green `ci/circleci` builds in the "All checks have passed" section of my PR ([connect CircleCI to GitHub](https://support.circleci.com/hc/en-us/articles/360008097173-Why-aren-t-pull-requests-triggering-jobs-on-my-organization-) if not)
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context
We want to fully automate device management on the Developer Portal, which includes renaming, enabling, and disabling devices.

### Description
Implemented App Store Connect APIs [Modify a Registered Device PATCH request](https://developer.apple.com/documentation/appstoreconnectapi/modify_a_registered_device).

### Testing Steps
```ruby
Spaceship::ConnectAPI::Device.disable("udid")
Spaceship::ConnectAPI::Device.enable("udid")
Spaceship::ConnectAPI::Device.rename("udid", "new name")
```

### CODE REVIEW

**Comment by getaaron** [comment]
Looks like `find_by_udid` doesn't actually need the `platform` param. I opened https://github.com/fastlane/fastlane/pull/20931 to remove it.

File: spaceship/lib/spaceship/connect_api/models/device.rb
Comment (nekrich) [reply spaceship/lib/spaceship/connect_api/models/device.rb:103]: No, it could not, cause it's if `enabled` is set, then `||=` will become `boolean or and assign` and will fail to disable the enabled device
```
var = false
var ||= true
puts var.to_s # true
```

**Comment by nekrich** [comment]
@getaaron , thanks for the review. Fixed everything I could 😅

File: spaceship/lib/spaceship/connect_api/models/device.rb
Comment (getaaron) [reply spaceship/lib/spaceship/connect_api/models/device.rb:102]: ohhh right 😄

**Comment by getaaron** [comment]
looks good! will merge after someone approves https://github.com/fastlane/fastlane/pull/20931

**Review by getaaron — APPROVED** [review]

**Comment by getaaron** [comment]
Thanks again for adding this!

**Comment by rasberik** [comment]
Can someone confirm if it still works, may be endpoints got updated?

I am getting authentication error on network response for enable and disable although `Spaceship::ConnectAPI::Device.all` is working as expected.

Had to use `Spaceship.device.all`(with same credentials) and then use `enable!` and `disable!` methods from `spaceship/lib/spaceship/portal/device.rb`, but unfortunately no renaming ability there

**Pull request merged**
